src/rust/ide/lib/view/src/graph_editor.rs

```diff
@@ -2178,8 +2207,14 @@ fn new_graph_editor(app:&Application) -> GraphEditor {
             let data    = Rc::new(sample_data_generator.generate_data()); // FIXME: why rc?
             let content = serde_json::to_value(data).unwrap();
             let data    = visualization::Data::from(content);
-            inputs.set_visualization_data.emit((*node_id,data));
+            inputs.set_visualization_data.emit((*node_id,data.clone()));
         }
+
+        let sample_doc_generator = MockDocGenerator::default();
+        let data    = sample_doc_generator.generate_data();
+        let content = serde_json::to_value(data).unwrap();
+        let data    = visualization::Data::from(content);
+        inputs.set_documentation_data.emit(data);
```

Comment:
As I see, this whole section should be removed (according to the TODO in line 2203. Please consult @wdanilo or @MichaelMauderer if it should be already removed (actually we send the "real" data right now).

Reply:
I'm not sure if it should at this moment, it is useful for now for testing purposes

Reply:
Yes, this section is most likely no longer really needed, but it can still be useful for testing.
It might be best to move this to a separate debug scene that has dummy data updates and some of this functionality might go there then.

Reply:
Yes, but it wouldn't be too adequate to do this with this PR, maybe next one which would also take care about the debug scenes

Reply:
It doesn't matter if its related to this PR or not. Part of this PR is delivering the functionality that is working correctly. If this is not needed here, it should be removed as @farmaazon and @MichaelMauderer suggested. Of course, it should also be integrated with debug scenes / shapes - every visual task has to BOTH work in GUI and debug scenes - this is requirement for merging anything visual. Mock data should definitely not be part of graph_editor per se.